server/src/frontend_api.rs

```diff
@@ -220,7 +220,11 @@ fn post_all_features(
     let engine = engine_cache.get(&key).ok_or_else(|| {
         EdgeError::FrontendNotYetHydrated(FrontendHydrationMissing::from(&edge_token))
     })?;
-    let feature_results = engine.resolve_all(&context_with_ip, &None).unwrap();
+    let feature_results = engine.resolve_all(&context_with_ip, &None).ok_or_else(|| {
+        EdgeError::FrontendExpectedToBeHydrated(
+            "Feature cache has not been hydrated yet, but it was expected to be".into(),
```

Comment:
Ok, maybe too verbose, but as a user this error would tell me nothing
```suggestion
            "Feature cache has not been hydrated yet, but it was expected to be. This can be due to a race condition from calling edge before it's ready. This error might auto resolve as soon as edge is able to fetch from upstream".into(),
```

Reply:
This isn't that error though, that's handled by Edge hydration in the line above. This is deep into "things that should never happen". The only path to this error happening is if Yggdrasil has been inserted into the compiled cache without having been compiled and that should now not happen.

The error log that points out the toggles are broken is logged in the state hydration handling code

But this looks silly. Lemme patch Yggdrasil so this special case goes away

Reply:
Nice, I took your suggestion. It makes sense now. The only way this can happen is if Ygg has not been hydrated now